gravitational/gravity
# [5.5.x] Add priority classes to dns, gravity and tiller apps

## Description

This PR sets high priority classes for Tiller deployment, gravity-site daemonset and coredns daemonset.

## Type of change

* New feature (non-breaking change which adds functionality)

## Linked tickets and other PRs



* Parially fixes https://github.com/gravitational/gravity/issues/1611 for 5.5

* Requires [rigging #78](https://github.com/gravitational/rigging/pull/78), [planet #681](https://github.com/gravitational/planet/pull/681), [logging #65](https://github.com/gravitational/logging-app/pull/65) and [monitoring #175](https://github.com/gravitational/monitoring-app/pull/175)

## TODOs


## Implementation


## Testing done

1. Install cluster with changes from this PR.
2. Create example nginx deployment.
```
kubectl apply -f https://k8s.io/examples/application/deployment.yaml
```
3. Set resource limits for nginx deployment.
```
kubectl set resources deployment nginx-deployment --limits=cpu=400m,memory=512Mi
```
4. Scale up nginx deployment to reach resources limits for nodes
```
kubectl scale deployment nginx-deployment --replicas=40
```
5. Check that some pods are in `Pending` state.
```
kubectl get pod -o wide | grep -i pending
nginx-deployment-76c987f6c4-464ft   0/1     Pending       0          31s     <none>         <none>         <none>           <none>
nginx-deployment-76c987f6c4-4wx6c   0/1     Pending       0          31s     <none>         <none>         <none>           <none>
nginx-deployment-76c987f6c4-8dhj9   0/1     Pending       0          31s     <none>         <none>         <none>           <none>
nginx-deployment-76c987f6c4-929c2   0/1     Pending       0          16s     <none>         <none>         <none>           <none>
```
6. Delete one of the coredns pods and check it is rescheduled again.
```
kubectl --namespace=kube-system delete pod coredns-fns46
pod "coredns-fns46" deleted

kubectl get pod -o wide --all-namespaces| grep cored
kube-system   coredns-9n9kn                         1/1     Running     0          11s     10.244.51.31   sergei-dev-2   <none>           <none>
kube-system   coredns-lgm8n                         1/1     Running     0          46m     10.244.24.2    sergei-dev-1   <none>           <none>
kube-system   coredns-v8vns                         1/1     Running     0          46m     10.244.95.3    10.162.0.23    <none>           <none>
```
7. Repeate the same step 6 for tiller and gravity-site pods.
## Additional information
